File: app/src/main/java/com/njlabs/showjava/ui/FilePickerFragment.java
package com.njlabs.showjava.ui;

import android.support.annotation.NonNull;

import java.io.File;

public class FilePickerFragment extends com.nononsenseapps.filepicker.FilePickerFragment {

    private static final String EXTENSION = ".apk";

    /**
     * @param file input file
     * @return The file extension. If file has no extension, it returns null.
     */
    private String getExtension(@NonNull File file) {
        String path = file.getPath();
        int i = path.lastIndexOf(".");
        if (i < 0) {
            return null;
        } else {
            return path.substring(i);
        }
    }

    @Override
    protected boolean isItemVisible(final File file) {
        // Default behavior
        // return isDir(file) || (mode == MODE_FILE || mode == MODE_FILE_AND_DIR);
        if (!isDir(file) && (mode == MODE_FILE || mode == MODE_FILE_AND_DIR)) {
            return EXTENSION.equalsIgnoreCase(getExtension(file));
        }
        return isDir(file);
    }

    /**
     * For consistency, the top level the back button checks against should be the start path.
     * But it will fall back on /.
     */
    private File getBackTop() {
        if (getArguments().containsKey(KEY_START_PATH)) {
            String keyStartPath = getArguments().getString(KEY_START_PATH);
            return getPath((keyStartPath != null ? keyStartPath : "/"));
        } else {
            return new File("/");
        }
    }

    /**
     * @return true if the current path is the startpath or /
     */
    public boolean isBackTop() {
        return 0 == compareFiles(mCurrentPath, getBackTop()) || 0 == compareFiles(mCurrentPath, new File("/"));
    }

    /**
     * Go up on level, same as pressing on "..".
     */
    public void goUp() {
        mCurrentPath = getParent(mCurrentPath);
        mCheckedItems.clear();
        mCheckedVisibleViewHolders.clear();
        refresh();
    }


}

File: app/src/main/kotlin/com/njlabs/showjava/activities/filepicker/FilePickerFragment.kt
package com.njlabs.showjava.activities.filepicker

import com.nononsenseapps.filepicker.AbstractFilePickerFragment
import java.io.File

class FilePickerFragment : com.nononsenseapps.filepicker.FilePickerFragment() {

    private val extensions = arrayOf(".apk", ".jar")

    /**
     * @param file input file
     * *
     * @return The file extension. If file has no extension, it returns null.
     */
    private fun getExtension(file: File): String? {
        val path = file.path
        val i = path.lastIndexOf(".")
        return if (i < 0) {
            null
        } else {
            path.substring(i)
        }
    }

    /**
     * Allow only selected extensions.
     */
    override fun isItemVisible(file: File?): Boolean {
        if (file != null) {
            if (!isDir(file) && (mode == MODE_FILE || mode == MODE_FILE_AND_DIR)) {
                return extensions.contains(getExtension(file)?.toLowerCase())
            }
            return isDir(file)
        }
        return false
    }

    /**
     * For consistency, the top level the back button checks against should be the start path.
     * But it will fall back on /.
     */
    private val backTop: File
        get() {
            return if (arguments!!.containsKey(AbstractFilePickerFragment.KEY_START_PATH)) {
                val keyStartPath = arguments!!.getString(AbstractFilePickerFragment.KEY_START_PATH)
                getPath(keyStartPath ?: "/")
            } else {
                File("/")
            }
        }

    /**
     * @return true if the current path is the start path or /
     */
    val isBackTop: Boolean
        get() = 0 == compareFiles(mCurrentPath, backTop) || 0 == compareFiles(mCurrentPath, File("/"))

    /**
     * Go up on level, same as pressing on "..".
     */
    override fun goUp() {
        mCurrentPath = getParent(mCurrentPath)
        mCheckedItems.clear()
        mCheckedVisibleViewHolders.clear()
        refresh(mCurrentPath)
    }
}
